purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1188,21 +1197,48 @@ class Purchases internal constructor(
 
     private fun getProducts(
         productIds: Set<String>,
-        productType: ProductType,
+        productTypes: Set<ProductType>,
         callback: GetStoreProductsCallback
     ) {
-        billing.queryProductDetailsAsync(
-            productType,
-            productIds,
-            { storeProducts ->
-                dispatch {
-                    callback.onReceived(storeProducts)
-                }
-            }, {
-                dispatch {
-                    callback.onError(it)
-                }
-            })
+       getProducts(
+           productIds,
+           productTypes,
+           emptyList(),
+           callback
+       )
+    }
+
+    private fun getProducts(
+        productIds: Set<String>,
+        productTypes: Set<ProductType>,
+        collectedStoreProducts: List<StoreProduct>,
+        callback: GetStoreProductsCallback
+    ) {
+        val productTypesLeft = productTypes.toMutableSet()
```

Comment:
not sure it's worth it, but maybe we can filter out the producttype UNKNOWN? i think we default to INAPP if it doesn't map correctly, but that'd be incorrect now that we expose the type as a parameter